public/video-ui/src/components/utils/GridImageSelect.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import Modal from './Modal';
+
+const GRID_URL = 'https://media.gutools.co.uk';
+
+export default class GridEmbedder extends React.Component {
+
+    state = {
+        modalOpen: false
+    }
+
+    toggleModal = () => {
+        if (this.state.modalOpen) {
+            this.closeModal();
+        } else {
+            this.openModal();
+        }
+    }
+
+    closeModal = () => {
+        this.setState({ modalOpen: false });
+        window.removeEventListener('message', this.onMessage, false);
+    }
+
+    openModal = () => {
+        this.setState({ modalOpen: true });
+        window.addEventListener('message', this.onMessage, false);
+    }
+
+    validMessage(data) {
+        return data && data.crop && data.crop.data && data.image && data.image.data;
```

Comment:
These data structures might be better parsed with https://github.com/argo-rest/theseus